### PR TITLE
[CI] Fix BWC related CI failures by swapping dist url to use dist min url instead.

### PR DESCRIPTION
**Pull request by manasvinibs**
### Description
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4814

Here I'm using min url instead of snapshot.

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4814

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: .github/workflows/build_and_test_workflow.yml [hunk 1]
```diff
@@ -346,7 +346,7 @@ jobs:
 
       - name: Set OpenSearch URL
         run: |
-          echo "OPENSEARCH_URL=https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ env.VERSION }}/latest/linux/x64/tar/dist/opensearch/opensearch-${{ env.VERSION }}-linux-x64.tar.gz" >> $GITHUB_ENV
+          echo "OPENSEARCH_URL=https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{ env.VERSION }}/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-${{ env.VERSION }}-linux-x64.tar.gz" >> $GITHUB_ENV
```
Comment (kavilla) [inline .github/workflows/build_and_test_workflow.yml:349]: I see the min is failing a different test now. The min for OpenSearch Dashboards is passed with a `--release` which compiles down our files to JS directly so any `dev` mode only stuff won't work on the `min`. I do not remember if the `min` is the same as the `snapshot` version for OpenSearch.

Quickly looking the link provided by the snapshot doesn't have the same metadata as the one on the min when I checked locally (but that isn't proof that they are different just suspect).

Do you have a reference to this pass on the 2.x branch that I can peek at?

**Comment by manasvinibs** [comment]
Currently this is failing with error https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/5969196554/job/16196895980?pr=4815 
Converting this to draft till we figure the next steps.

**Comment by kavilla** [comment]
@manasvinibs should we close this one in favor of the other one being merged into 2.x?

**Comment by manasvinibs** [comment]
> @manasvinibs should we close this one in favor of the other one being merged into 2.x?

Yes, I'll close this one and forward port the change that went into 2.x.

**Pull request closed without merging**